sympy/polys/matrices/domainmatrix.py

```diff
@@ -21,6 +21,47 @@
 
 
 class DomainMatrix:
+    r"""
+        The idea behind DomainMatrix is to associate :py:class:`~.Domain` with matrix.
+
+        Explanation
+        ===========
+
+        DomainMatrix uses "domain elements" for its internal representation
+        which makes it more faster for many common operations
+        than current sympy Matrix class, but this advantage makes it not
+        entirely compatible with Matrix.
+        DomainMatrix could be found analogous to numpy arrays with "dtype".
+        In the DomainMatrix, each matrix has a domain such as ZZ(Integers)
+        or QQ<sqrt(2)>[Algebraic Number Field].
+
+        Examples
+        ========
+
+        Creating a DomainMatrix from the existing Matrix class:
+
+        >>> from sympy import Matrix
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> Matrix1 = Matrix([
+        ...         [2, 73],
+        ...         [27, 86]])
+        >>> domainMatrix = DomainMatrix.from_Matrix(Matrix1)
+        >>> domainMatrix
+        DomainMatrix([[2, 73], [27, 86]], (2, 2), ZZ)
+
+        See Also
+        ========
+
+        :py:class:`~.ddm`
+        :py:class:`~.Domain`
+
+        References
+        ==========
+
+        .. module:: sympy.polys.domainsintro
+        :py:class:`~.Poly`
```

Comment:
References is for external references. See the style guide:
https://docs.sympy.org/latest/documentation-style-guide.html#references-section

Reply:
A link to `domainsintro` would make sense in the explanation section.

Reply:
Although really just linking to Domain is enough as that takes you here:
https://docs.sympy.org/dev/modules/polys/domainsref.html#sympy.polys.domains.domain.Domain